### ISSUE TITLE
Preloading / Lazy loading model before generate requested

### ISSUE THREAD

**Issue by gidzr** [xenova/transformers.js#528]
### Question

Hi @xenova 

I've been looking around for this type of functionality for ages and didn't realize you had this type of front-end inferencing locked down in such awesome fashion on browsers. Brilliant!!!

In the demo at https://xenova.github.io/transformers.js/, the model is loaded one-time when sending the first request/inference. 

I want to pre-load a model in the background when a user opens the page, but not sure on the whether there is a method in your API for https://cdn.jsdelivr.net/npm/@xenova/transformers@2.14.0, or whether model loading is purely contingent on a first inference.

I've checked your API link: https://huggingface.co/docs/transformers.js/api/env, and nothing there that I can see so I'm assuming it requires a first run.

If it requires a first-run I can think of a couple workarounds, but wanted to check with you before heading down that rabbit hole.

Cheers

**Comment by xenova** [comment]
Hi there 👋 The model will be loaded whenever you call `await pipeline(...)` or `AutoModelXXX.from_pretrained(...)`. In the demo, I wrap these calls in a Singleton, which allows me to lazily load the models. I've found this to provide the best user experience, as it avoids loading of large files on page load.

**Comment by gidzr** [comment]
Awesome, thanks!

FYI - I tested a few browser based LLM deployments/projects but found they were usually very slow and very limited. Things like webllm are great projects, but more proof of concept as they not flexible for real-world re-use in other projects. Then I tested out projects that create node.js testing environments in the browser (ie. browser as an IDE) - which was overkill and struggled to meet the use-case. There's almost an obsession with gpt and llama like models being flavour of the month, which completely overlooks real world requirements for the vast range of LLM / AI model applications. 

The key requirements I've been looking for (in a viable front-end LLM) are: LLM deployment to a browser front-end assuming **no gpu**, broad array of models or flexibility to custom switch models from same family, and pre-loading/caching of model. 

Your project is definitely doing this. Kudos!!
